pinot-tools/src/main/java/com/linkedin/pinot/tools/AutoAddInvertedIndex.java

```diff
@@ -0,0 +1,212 @@
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.apache.helix.PropertyPathConfig;
+import org.apache.helix.PropertyType;
+import org.apache.helix.ZNRecord;
+import org.apache.helix.manager.zk.ZKHelixAdmin;
+import org.apache.helix.manager.zk.ZNRecordSerializer;
+import org.apache.helix.store.zk.ZkHelixPropertyStore;
+import org.json.JSONObject;
+
+
+/**
+ * This tool will directly modify the table configs in ZooKeeper.
+ * PLEASE USE IT VERY CAREFULLY!
+ */
+public class AutoAddInvertedIndex {
+  private AutoAddInvertedIndex() {
+  }
+
+  private static final String AUTO_GENERATED_INVERTED_INDEX_KEY = "autoGeneratedInvertedIndex";
+
+  private static final String ZK_ADDRESS = "zkAddress:zkPort";
+  private static final String CLUSTER_NAME = "pinot-cluster";
+  private static final String TABLE_CONFIG_PATH = "/CONFIGS/TABLE/";
+  private static final String SCHEMA_PATH = "/SCHEMAS/";
+  private static final ZKHelixAdmin HELIX_ADMIN = new ZKHelixAdmin(ZK_ADDRESS);
+  private static final ZkHelixPropertyStore<ZNRecord> PROPERTY_STORE =
+      new ZkHelixPropertyStore<>(ZK_ADDRESS, new ZNRecordSerializer(),
+          PropertyPathConfig.getPath(PropertyType.PROPERTYSTORE, CLUSTER_NAME));
+
+  private static final String BROKER_QUERY_URL = "http://brokerAddress:brokerPort/query";
+
+  private static final long TABLE_SIZE_THRESHOLD = 100_000_000;
+
+  private static AbstractTableConfig getTableConfig(String tableName)
+      throws Exception {
+    return AbstractTableConfig.fromZnRecord(PROPERTY_STORE.get(TABLE_CONFIG_PATH + tableName, null, 0));
+  }
+
+  private static void setTableConfig(String tableName, AbstractTableConfig tableConfig)
+      throws Exception {
+    ZNRecord znRecord = AbstractTableConfig.toZnRecord(tableConfig);
+    znRecord.setSimpleField(AUTO_GENERATED_INVERTED_INDEX_KEY, "true");
+    PROPERTY_STORE.set(TABLE_CONFIG_PATH + tableName, znRecord, 1);
+  }
+
+  private static Schema getTableSchema(String tableName)
+      throws Exception {
+    ZNRecord znRecord = PROPERTY_STORE.get(SCHEMA_PATH + tableName, null, 0);
+    if (znRecord == null) {
+      return null;
+    } else {
+      return SchemaUtils.fromZNRecord(znRecord);
+    }
+  }
+
+  private static JSONObject sendQuery(String query)
+      throws Exception {
+    URLConnection urlConnection = new URL(BROKER_QUERY_URL).openConnection();
+    urlConnection.setDoOutput(true);
+
+    BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(urlConnection.getOutputStream(), "UTF-8"));
+    writer.write(new JSONObject().put("pql", query).toString());
+    writer.flush();
+
+    BufferedReader reader = new BufferedReader(new InputStreamReader(urlConnection.getInputStream(), "UTF-8"));
+    return new JSONObject(reader.readLine());
+  }
+
+  private static class ResultPair implements Comparable<ResultPair> {
```

Comment:
ImmutablePair is a final class. I want to make this pair comparable, and override the toString() method to print useful information.